src/vs/workbench/contrib/tasks/browser/abstractTaskService.ts

```diff
@@ -607,7 +623,9 @@ export abstract class AbstractTaskService extends Disposable implements ITaskSer
 	}
 
 	protected _disposeTaskSystemListeners(): void {
-		this._taskSystemListener?.dispose();
+		if (this._taskSystemListeners) {
+			dispose(this._taskSystemListeners);
```

Comment:
`this._taskSystemListeners = undefined` to free the objects before refs to this are cleared?

Reply:
shouldn't that be done after the above?

Reply:
Yes, I mean:

```ts
if (this._taskSystemListeners) {
  dispose(this._taskSystemListeners);
  // Setting to undefined here will free this reference, otherwise the objects in
  // _taskSystemListeners will stick around until the references to `this` are freed
  this._taskSystemListeners = undefined;
}
```